datashader/glyphs/points.py

```diff
@@ -72,11 +77,23 @@ def required_columns(self):
         return [self.geometry]
 
     def compute_x_bounds(self, df):
-        bounds = df[self.geometry].array.total_bounds_x
+        col = df[self.geometry]
+        if isinstance(col.dtype, gpd_GeometryDtype):
+            # geopandas
+            bounds = col.total_bounds[::2]
```

Comment:
One small optimization would be to calculate the total_bounds only once (and not twice for x bounds and y bounds separately), if that's doable in the current code organisation

Reply:
Yes, I can do this.

Reply:
Done